openquake/commonlib/calculators/base.py

```diff
@@ -70,7 +71,8 @@ def __init__(self, oqparam, monitor=DummyMonitor(), calc_id=None,
         else:
             self.datastore = general.AccumDict()
             self.datastore.hdf5 = {}
-        self.oqparam = oqparam
+        if 'oqparam' not in self.datastore:  # new datastore
+            self.oqparam = oqparam
```

Comment:
I checked. This is actually correct. I have added a comment explaining that this is the case when there is a precalculator and oqparam has been already stored in the datastore.